lib/Espfc/src/Device/InputESPNOW.h

```diff
@@ -0,0 +1,187 @@
+#ifndef _INPUT_DEVICE_INPUT_ESPNOW_H_
+#define _INPUT_DEVICE_INPUT_ESPNOW_H_
+
+#include "esp_wifi.h"
+#include <WiFi.h>
+#include <esp_now.h>
+#include "Device/InputDevice.h"
+#include "Math/Utils.h"
+
+
+
+namespace Espfc {
+
+namespace Device {
+
+struct ControllerData {
+  uint8_t syncByte: 1;
+  uint16_t A: 10;
+  uint16_t E: 10;
+  uint16_t R: 10;
+  uint16_t T: 10;
+
+  uint8_t ch1;
+  uint8_t ch2;
+  uint8_t ch3;
+  uint8_t ch4;
+}_data;
+
+int8_t rssi;
+
+struct TelemetryData {
+  uint8_t syncByte: 1;
+
+  uint8_t Battery_v; // int/10 -> fixed point 25.6v max
+}TxData;
+
+//unsigned long lastRecvTime = 0;
+bool _new_data = false;
+
+#define wifi_channel 4 // Set the wifi channel (1-13) -> TODO: scan and choose the best one
+uint8_t TxMacAddr[] = {0xA8,0x42,0xE3,0xCD,0x5F,0x04}; // REPLACE WITH YOUR RECEIVER MAC Address -> TODO: put it into model
+bool pair_status = false; //TODO: should be save in model, can be changed using cil
```

Comment:
as you mention, it would be good to have these data in model, and it would perfect to pass them via begin function, and if change is required (after pairing), pass them via callback function to caller. In this way we get a generic solution with dependency isolation.